addons/mail/static/src/messaging/entity/partner/partner.js

```diff
@@ -172,49 +209,48 @@ function PartnerFactory({ Entity }) {
         }
 
         /**
-         * @override
+         * @private
+         * @returns {string}
          */
-        _createInstanceLocalId(data) {
-            return `${this.constructor.name}_${data.id}`;
+        _computeNameOrDisplayName() {
+            return this.name || this.display_name;
         }
 
         /**
          * @override
          */
-        _update(data) {
-            const {
-                display_name = this.display_name || "",
-                email = this.email,
-                id = this.id,
-                im_status = this.im_status,
-                name = this.name,
-                userId,
-            } = data;
-
-            Object.assign(this, {
-                display_name,
-                email,
-                id,
-                im_status,
-                model: 'res.partner',
-                name,
-            });
-
-            if (userId) {
-                const user = this.env.entities.User.insert({ id: userId });
-                this.link({ user });
-            }
+        _createInstanceLocalId(data) {
+            return `${this.constructor.entityName}_${data.id}`;
         }
 
     }
 
+    Partner.entityName = 'Partner';
+
     Partner.fields = {
         directPartnerThread: one2one('Thread', {
             inverse: 'directPartner',
         }),
+        display_name: attr({
+            default: "",
```

Comment:
Why default this one in particular to string? And not for example email?
Feels like field type should handle it eventually.